### PR TITLE
Document some of the environment variables used by fwupd

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd/issues/3760

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [x] Documentation

### CODE REVIEW

File: docs/env.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+* `FU_HID_DEVICE_VERBOSE` shows HID traffic
+* `FU_SREC_FIRMWARE_VERBOSE` shows more information about parsing
+* `FU_UDEV_DEVICE_DEBUG` shows more information about UDEV devices, including parents
+* `FU_USB_DEVICE_DEBUG` shows more information about USB devices
+* `FWUPD_DEVICE_LIST_VERBOSE` display devices being added and removed from the list
+* `FWUPD_PROBE_VERBOSE` dump the detected devices to the console, even if not supported by fwupd
+
+## Plugins
+
+Most plugins read a plugin-specific key to increase verbosity, e.g. `FWUPD_DELL_VERBOSE`.
```
Comment (superm1) [inline docs/env.md:33]: I think explaining how the option for plugin verbose sets this makes sense

**Review by superm1 — COMMENTED** [review]
missing proxy variables for HTTPS/HTTP

**Comment by superm1** [comment]
I think probably should be referenced from `fwupdplugin.toml.in` and/or `fwupd.toml.in` right?

File: docs/env.md (hunk 1 above)
Comment (hughsie) [reply docs/env.md:33]: Added

**Comment by hughsie** [comment]
> missing proxy variables for HTTPS/HTTP, and also LANG variable

They're not really self-consumed variables, but I guess they're useful and I've briefly documented them.

> I think probably should be referenced from fwupdplugin.toml.in and/or fwupd.toml.in right?

I'm not sure. fwupd docs are aimed at *I'm building a GUI for fwpud* and fwupdplugin is aimed at `I'm building a plugin for fwupd` and we don't really have a website aimed at end users wanting to debug things.I suppose it's leaning more on the 2nd, although a markdown document seems a safe first step.

**Comment by superm1** [comment]
>I'm not sure. fwupd docs are aimed at I'm building a GUI for fwpud and fwupdplugin is aimed at I'm building a plugin for fwupd and we don't really have a website aimed at end users wanting to debug things.I suppose it's leaning more on the 2nd, although a markdown document seems a safe first step.

I was thinking the second as well.  The logic being most of the people using this will come from the plugin development side to figure out how to debug problems.  So if you put it near the tutorial (and probably also link from the tutorial) that might help.

**Review by superm1 — DISMISSED** [review]
LGTM, I think discoverability needs to go up.

**Comment by hughsie** [comment]
> I think discoverability needs to go up.

I found and fixed the bug which caused the titles to not show up on the index, and so included the env vars in the docs.

**Pull request merged**